pennylane_qiskit/converter.py

```diff
@@ -328,56 +327,62 @@ def _function(*args, params: dict = None, wires: list = None, **kwargs):
-                # Extract the bound parameters from the operation. If the bound parameters are a
-                # Qiskit ParameterExpression, then replace it with the corresponding PennyLane
-                # variable from the unbound_params dictionary.
-
-                pl_parameters = []
-                for p in op.params:
-                    _check_parameter_bound(p, unbound_params)
-
-                    if isinstance(p, ParameterExpression):
-                        if p.parameters:  # non-empty set = has unbound parameters
-                            ordered_params = tuple(p.parameters)
-
-                            f = lambdify(ordered_params, p._symbol_expr, modules=qml.numpy)
-                            f_args = []
-                            for i_ordered_params in ordered_params:
-                                f_args.append(unbound_params.get(i_ordered_params))
-                            pl_parameters.append(f(*f_args))
-                        else:  # needed for qiskit<0.43.1
-                            pl_parameters.append(float(p))  # pragma: no cover
-                    else:
-                        pl_parameters.append(p)
-
-                execute_supported_operation(
-                    inv_map[instruction_name], pl_parameters, operation_wires
-                )
+            # Define operator builders and helpers
+            operation_func = None
 
-            elif instruction_name in dagger_map:
-                gate = dagger_map[instruction_name]
-                qml.adjoint(gate)(wires=operation_wires)
+            def operation_overlapper(op):
+                return op
 
-            elif isinstance(op, Measure):
+            operation_wires = [wire_map[hash(qubit)] for qubit in qargs]
+            operation_kwargs = {"wires": operation_wires}
+            operation_args = []
+            operation_cond = False
+
+            # Extract the bound parameters from the operation. If the bound parameters are a
+            # Qiskit ParameterExpression, then replace it with the corresponding PennyLane
+            # variable from the var_ref_map dictionary.
+            operation_params = []
+            for p in ops.params:
+                _check_parameter_bound(p, unbound_params)
+
+                if isinstance(p, ParameterExpression):
+                    if p.parameters:  # non-empty set = has unbound parameters
+                        ordered_params = tuple(p.parameters)
+                        f = lambdify(ordered_params, p._symbol_expr, modules=qml.numpy)
+                        f_args = []
+                        for i_ordered_params in ordered_params:
+                            f_args.append(unbound_params.get(i_ordered_params))
+                        operation_params.append(f(*f_args))
+                    else:  # needed for qiskit<0.43.1
+                        operation_params.append(float(p))  # pragma: no cover
+                else:
+                    operation_params.append(p)
```

Comment:
Looks like this could be extracted into a helper function `operation_params = _get_operation_params(ops)`.